### PR TITLE
Resolve #975 Pull Request Confilct with additional unit testing processes

### PR DESCRIPTION
**Pull request by sed1ka**
Reference by #975 with additional unit testing processes

But while doing unit testing, I found a strange trait in using VerticalLine in BarChartData
In  VerticalLine we need to send the value of X, but BarChartData uses a different calculation to get the value of X for each data (BarChartGroupData). So, for now, the VerticalLine in BarChartData can only be set within a range of 0 to 1 for the line to appear.

I need extraLine in BarChart ASAP, so I haven't solved that strange trait

### CODE REVIEW

File: lib/src/chart/bar_chart/bar_chart_data.dart [hunk 1]
```diff
@@ -115,37 +143,14 @@ class BarChartData extends AxisChartData with EquatableMixin {
     );
   }
 
-  /// Lerps a [BaseChartData] based on [t] value, check [Tween.lerp].
-  @override
```
Comment (imaNNeoFighT) [inline lib/src/chart/bar_chart/bar_chart_data.dart:119]: It seems these are unwanted changes. Please revert them.

File: lib/src/chart/bar_chart/bar_chart_data.dart [hunk 2]
```diff
@@ -253,7 +258,7 @@ class BarChartGroupData with EquatableMixin {
   static BarChartGroupData lerp(
       BarChartGroupData a, BarChartGroupData b, double t) {
     return BarChartGroupData(
-      x: (a.x + (b.x - a.x) * t).round(),
+      x: (a.x + (b.x - a.x) * t + 50).round(),
```
Comment (imaNNeoFighT) [inline lib/src/chart/bar_chart/bar_chart_data.dart:261]: What is `+ 50` for?

File: lib/src/chart/base/axis_chart/axis_chart_data.dart [hunk 3]
```diff
@@ -230,8 +236,7 @@ class AxisTitles with EquatableMixin {
   bool get showAxisTitles => axisNameWidget != null && axisNameSize != 0;
 
   /// If there is something to show as sideTitles, it returns true
-  bool get showSideTitles =>
-      sideTitles.showTitles && sideTitles.reservedSize != 0;
+  bool get showSideTitles => sideTitles.showTitles && sideTitles.reservedSize != 0;
```
Comment (imaNNeoFighT) [inline lib/src/chart/base/axis_chart/axis_chart_data.dart:239]: Unnecessary change.

File: lib/src/chart/base/axis_chart/axis_chart_data.dart [hunk 4]
```diff
@@ -519,8 +524,7 @@ class FlGridData with EquatableMixin {
     return FlGridData(
       show: b.show,
       drawHorizontalLine: b.drawHorizontalLine,
-      horizontalInterval:
-          lerpDouble(a.horizontalInterval, b.horizontalInterval, t),
+      horizontalInterval: lerpDouble(a.horizontalInterval, b.horizontalInterval, t),
```
Comment (imaNNeoFighT) [inline lib/src/chart/base/axis_chart/axis_chart_data.dart:527]: Unnecessary change.

File: lib/src/chart/base/axis_chart/axis_chart_data.dart [hunk 5]
```diff
@@ -547,16 +551,12 @@ class FlGridData with EquatableMixin {
       show: show ?? this.show,
       drawHorizontalLine: drawHorizontalLine ?? this.drawHorizontalLine,
       horizontalInterval: horizontalInterval ?? this.horizontalInterval,
-      getDrawingHorizontalLine:
-          getDrawingHorizontalLine ?? this.getDrawingHorizontalLine,
-      checkToShowHorizontalLine:
-          checkToShowHorizontalLine ?? this.checkToShowHorizontalLine,
+      getDrawingHorizontalLine: getDrawingHorizontalLine ?? this.getDrawingHorizontalLine,
```
Comment (imaNNeoFighT) [inline lib/src/chart/base/axis_chart/axis_chart_data.dart:554]: Unnecessary change.

File: lib/src/chart/base/axis_chart/axis_chart_data.dart [hunk 6]
```diff
@@ -547,16 +551,12 @@ class FlGridData with EquatableMixin {
...
+      checkToShowHorizontalLine: checkToShowHorizontalLine ?? this.checkToShowHorizontalLine,
       drawVerticalLine: drawVerticalLine ?? this.drawVerticalLine,
       verticalInterval: verticalInterval ?? this.verticalInterval,
-      getDrawingVerticalLine:
-          getDrawingVerticalLine ?? this.getDrawingVerticalLine,
-      checkToShowVerticalLine:
-          checkToShowVerticalLine ?? this.checkToShowVerticalLine,
+      getDrawingVerticalLine: getDrawingVerticalLine ?? this.getDrawingVerticalLine,
```
Comment (imaNNeoFighT) [inline lib/src/chart/base/axis_chart/axis_chart_data.dart:558]: Unnecessary change.

File: lib/src/chart/base/axis_chart/axis_chart_data.dart [hunk 7]
```diff
@@ -702,13 +702,12 @@ class RangeAnnotations with EquatableMixin {
...
   /// Lerps a [RangeAnnotations] based on [t] value, check [Tween.lerp].
-  static RangeAnnotations lerp(
-      RangeAnnotations a, RangeAnnotations b, double t) {
+  static RangeAnnotations lerp(RangeAnnotations a, RangeAnnotations b, double t) {
     return RangeAnnotations(
-      horizontalRangeAnnotations: lerpHorizontalRangeAnnotationList(
-          a.horizontalRangeAnnotations, b.horizontalRangeAnnotations, t),
-      verticalRangeAnnotations: lerpVerticalRangeAnnotationList(
-          a.verticalRangeAnnotations, b.verticalRangeAnnotations, t),
+      horizontalRangeAnnotations:
```
Comment (imaNNeoFighT) [inline lib/src/chart/base/axis_chart/axis_chart_data.dart:707]: Unnecessary changes

File: lib/src/chart/base/axis_chart/axis_chart_data.dart [hunk 8]
```diff
@@ -719,10 +718,8 @@ class RangeAnnotations with EquatableMixin {
     List<VerticalRangeAnnotation>? verticalRangeAnnotations,
   }) {
     return RangeAnnotations(
-      horizontalRangeAnnotations:
-          horizontalRangeAnnotations ?? this.horizontalRangeAnnotations,
-      verticalRangeAnnotations:
-          verticalRangeAnnotations ?? this.verticalRangeAnnotations,
+      horizontalRangeAnnotations: horizontalRangeAnnotations ?? this.horizontalRangeAnnotations,
```
Comment (imaNNeoFighT) [inline lib/src/chart/base/axis_chart/axis_chart_data.dart:721]: Unnecessary changes

**Comment by imaNNeoFighT** [comment]
There are a lot of unnecessary changes. Please revert them. I will review it again.

**Comment by imaNNeoFighT** [comment]
BTW #975 is exactly what you did here.

File: lib/src/chart/base/axis_chart/axis_chart_data.dart (hunk 3 above)
Comment (sed1ka) [reply lib/src/chart/base/axis_chart/axis_chart_data.dart]: Previously I used the darts code style line length to be 120 and forgot to change it to default (80) while working on this. So when I use auto wrapping, that's what happens. Please review my latest changes. Thankyou

**Comment by imaNNeoFighT** [comment]
You need to write some unit-tests

**Comment by sed1ka** [comment]
@imaNNeoFighT Didn't I already make a unit test?

**Comment by imaNNeoFighT** [comment]
Please look at the codecov result [here](https://github.com/imaNNeoFighT/fl_chart/pull/1110#issuecomment-1205154284).
